### PR TITLE
Bux-398/ get waiting contacts

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
Added GetContacts action, 
removed unused files, 
change to ShouldBindJson method in contact creation, 
Optimized newContact arguments, 
adjusted ContactService interface, 
changed id generation in contact model, 
added tests, 
added getContacts handler
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: actions/contacts/create.go [hunk 1]
```diff
@@ -23,13 +25,19 @@ import (
...
-	fullName := c.GetString("full_name")
-	paymail := c.GetString("paymail")
-	pubKey := c.GetString("pubKey")
+	var contactStruct engine.Contact
+
+	err := c.ShouldBindJSON(&contactStruct)
+
+	if err != nil {
+		fmt.Println("err", err)
+	}
```
Comment (chris-4chain) [inline actions/contacts/create.go:34]: I think, fmt.Println should be replaced to use a logger or maybe the printing is not necessary here?

File: actions/contacts/create.go [hunk 2]
```diff
@@ -23,13 +25,19 @@ import (
 // @Security	bux-auth-xpub
 func (a *Action) create(c *gin.Context) {
 
-	fullName := c.GetString("full_name")
-	paymail := c.GetString("paymail")
-	pubKey := c.GetString("pubKey")
+	var contactStruct engine.Contact
```
Comment (chris-4chain) [inline actions/contacts/create.go:28]: Consider another name without "Struct" suffix.

File: actions/contacts/create.go (hunk 1 above)
Comment (Nazarii-4chain) [reply actions/contacts/create.go]: Changed

File: actions/contacts/create.go (hunk 2 above)
Comment (Nazarii-4chain) [reply actions/contacts/create.go:27]: Changed

File: engine/model_contact_test.go [hunk 3]
```diff
@@ -102,3 +96,62 @@ func Test_getContact(t *testing.T) {
...
+		defer deferMe()
+
+		var metadata *Metadata
+
+		dbConditions := make(map[string]interface{})
+
+		var queryParams *datastore.QueryParams
+
+		(dbConditions)[xPubIDField] = xPubID
+		(dbConditions)[contactStatus] = notConfirmed
```
Comment (chris-4chain) [inline engine/model_contact_test.go:112]: Is it necessary to put the `dbConditions` variable into parenthesis?

Comment (Nazarii-4chain) [reply engine/model_contact_test.go]: Changed. Thanks

File: actions/contacts/create.go (hunk 2 above)
Comment (chris-4chain) [reply actions/contacts/create.go:27]: I can't see that you've changed it :sweat_smile:

Comment (Nazarii-4chain) [reply actions/contacts/create.go:27]: Changed in another file LoL

File: actions/contacts/search.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+// @Produce		json
+// @Param		page query int false "page"
+// @Param		page_size query int false "page_size"
+// @Param		order_by_field query string false "order_by_field"
+// @Param		sort_direction query string false "sort_direction"
+// @Param		conditions query string false "conditions"
+// @Success		200
+// @Router		/v1/contacts [get]
+// @Security	x-auth-xpub
+func (a *Action) search(c *gin.Context) {
```
Comment (chris-4chain) [inline actions/contacts/search.go:28]: 1. Will this `search` support querying for substrings for a autosuggestion feature?

image

2. Does it support fetching all the contacts at once? - maybe if I don't provide page & page_size params?
3. Is the searching by params case sensitive? I'm wondering what should happen when a user have a contact "BoB@b.com" and he puts into paymail field "bob@b.com". I suppose the app should inform a user that it is a trusted contact.
image

Comment (arkadiuszos4chain) [reply actions/contacts/search.go:28]: 1. Good question. In my opinion, we should be consistent with other `/search` endpoints in the system for now.
2. Leave `page` and `page_size` as optional parameters for consistency, but DO NOT set any MAX_PAGE_SIZE on server side for now.
3. We sanitize paymails (at least we should, I'll ensure we do it during adding contact), so we should make sure the paymail condition is in lower case to not rely on database collation.

Comment (Nazarii-4chain) [reply actions/contacts/search.go:28]: > 1. Will this `search` support querying for substrings for a autosuggestion feature?
> 
> image
> 
> 2. Does it support fetching all the contacts at once? - maybe if I don't provide page & page_size params?
> 3. Is the searching by params case sensitive? I'm wondering what should happen when a user have a contact "[BoB@b.com](mailto:BoB@b.com)" and he puts into paymail field "[bob@b.com](mailto:bob@b.com)". I suppose the app should inform a user that it is a trusted contact. Or maybe the "paymail" should be a `special` field and it should always be transformed to lower case.
>    image

1. It should work. It has the same logic under the hood as other `search`s. 
2. Yes. If you don't provide any metadata and params it should fetch all contacts.
3. Will add sanitizing paymails too

File: engine/model_contact_test.go (hunk 3 above)
Comment (chris-4chain) [reply engine/model_contact_test.go]: During review I've noticed that you haven'tt changed it in all places where the `(dbConditions)[` occurs. Please adjust.

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**